### DISCUSSION TITLE
How to add image attribute manually?

### DISCUSSION THREAD

**Discussion by youth-flutter-dev** [singerdmx/flutter-quill#688] [General]
I have image link, and I would like add that image link to quill document as image attribute as below.

```
{
      "insert": {
         "image": "https://user-images.githubusercontent.com/122956/72955931-ccc07900-3d52-11ea-89b1-d468a6e2aa2b.png"
      },
      "attributes":{
         "style":"mobileWidth: 50; mobileHeight: 50; mobileMargin: 10; mobileAlignment: topLeft"
      }
}
```

So How can I add image link to document as image attribute manually?

**Comment by youth-flutter-dev** [comment]
```
String emojiImage =
          'https://cdn.jsdelivr.net/npm/emoji-datasource-apple/img/apple/64/${unifiedEmoji}.png';
final int index = insertQuillController!.selection.baseOffset;
final int length = insertQuillController!.selection.extentOffset - index;
insertQuillController!.replaceText(index, length, BlockEmbed.image(emojiImage), null);
```

Add image with image attribute, it is ok now, but after inserting image, the cursor position isn't correct position. it is showing earlier image inserted position.

for example

=>ABCDE[Cursor]
=>ABCDE[AddedImage] 
=>ABCDE[Cursor][AddedImage] 

after adding some text again, it shows cursor in correct position;

=>ABCDE[AddedImage]FG[Cursor]

How can I solve this issue?

Could you please kindly check it? 
Thank you

**Reply by CatHood0** [reply]
Idk if you already solve this issue, but i could have a simple answer for this. You just need to update the cursor position after insert any custom block embed.

You can use this to move the position after the image:
```dart
controller.updateSelection(
     TextSelection.collapsed(
          offset: controller.selection.extentOffset + 1,
     ),
     ChangeSource.local,
);
```

**Comment by rakib205** [comment]
facing the same issue on the web.
Any fixes?

**Comment by adinghub** [comment]
Hello, I'm having your question now, how do you add image attributes? Much appreciated
